syntax = "proto3";
package nanoswap.message;

import "google/protobuf/timestamp.proto";


message LoanPayment {
    string payment_id = 1;
    int64 amount_due = 2;
    google.protobuf.Timestamp due_date = 3;
    optional string transaction = 4; // for completed payments, this is the transaction id of the payment
}


// ipfs filename:
//   bid/borrower_<id>.lender_<id>/loan_<id>/created_<timestamp>
message Loan {
    int64 principal_amount = 1;
    repeated LoanPayment repayment_schedule = 2;
    google.protobuf.Timestamp offer_expiry = 3;
    bool accepted = 4; // accepted timestamp must be before offer_expiry
    string lender_deposit_wallet = 5;
    string borrower_deposit_wallet = 6;
}
